### PR TITLE
Push Notifications: API integration mock

### PR DESCRIPTION
**Pull request by simonrozsival**
## What's this?
This PR adds the API token subscription/unsubscription requests mocks. It will allow us to implement our side of the app before the BE team implements the endpoints.

### Relationships

Ref #5033 

## Why do we want this?
:sanic:

## How is it done?
The endpoints return `Unit` observable for now and they don't make any actual HTTP requests.

### Side effects
None.

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Networking/ApiClients/PushServicesApi.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            Endpoints endPoints,
+            IApiClient apiClient,
+            IJsonSerializer serializer,
+            Credentials credentials)
+            : base(apiClient, serializer, credentials, endPoints.LoggedIn)
+        {
+            this.endPoints = endPoints.PushServices;
+        }
+
+        public IObservable<Unit> Subscribe(PushNotificationsToken token)
```
Comment (heytherewill) [inline Toggl.Networking/ApiClients/PushServicesApi.cs]: Shouldn't this be a boolean to indicate success/failure?

File: Toggl.Shared/PushNotificationsToken.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+namespace Toggl.Shared
+{
+    public struct PushNotificationsToken
+    {
+        private readonly string token;
+
+        public PushNotificationsToken(string token)
+        {
+            Ensure.Argument.IsNotNullOrEmpty(token, nameof(token));
```
Comment (heytherewill) [inline Toggl.Shared/PushNotificationsToken.cs]: Do we know what the token looks like? Some validation here would be nice

Comment (simonrozsival) [reply Toggl.Shared/PushNotificationsToken.cs]: The token is whatever the library gives us, it's not user input, I don't want to have any validation here.

File: Toggl.Networking/ApiClients/PushServicesApi.cs (hunk 1 above)
Comment (simonrozsival) [reply Toggl.Networking/ApiClients/PushServicesApi.cs]: The observable will throw if the HTTP request fails.

**Pull request merged**